tests/Unit/Helpers/Domain/DomainTestHelpers.hpp

```diff
@@ -54,7 +57,11 @@ void test_domain_construction(
         functions_of_time = {},
     const std::vector<std::unique_ptr<
         domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, VolumeDim>>>&
-        expected_grid_to_inertial_maps = {}) noexcept;
+        expected_grid_to_inertial_maps = {},
+    const std::vector<DirectionMap<
+        VolumeDim,
+        std::unique_ptr<domain::BoundaryConditions::BoundaryCondition>>>&
+        expected_boundary_conditions = {});
```

Comment:
Please add a note to the docs of this function that these expected boundary conditions as well as then domain's boundary conditions are expected to derive off `TestHelpers::domain::BoundaryConditions::TestBoundaryCondition`.